### PR TITLE
[RFC] vim-patch:7.4.706

### PR DESCRIPTION
**Pull request by buztard**
```
Problem:    Window drawn wrong when 'laststatus' is zero and there is a
            command-line window. (Yclept Nemo)
Solution:   Set the status height a bit later. (Christian Brabandt)
```

https://github.com/vim/vim/commit/v7-4-706

Original patch:

``` diff
diff --git a/src/nvim/version.c b/src/nvim/version.c
--- a/src/nvim/version.c
+++ b/src/nvim/version.c
@@ -742,6 +742,8 @@ static char *(features[]) =
 static int included_patches[] =
 {   /* Add new patch number below this line */
 /**/
+    706,
+/**/
     705,
 /**/
     704,
diff --git a/src/nvim/window.c b/src/nvim/window.c
--- a/src/nvim/window.c
+++ b/src/nvim/window.c
@@ -1236,7 +1236,8 @@ win_split_ins(size, flags, new_wp, dir)
    {
        wp->w_winrow = oldwin->w_winrow + oldwin->w_height + STATUS_HEIGHT;
        wp->w_status_height = oldwin->w_status_height;
-       oldwin->w_status_height = STATUS_HEIGHT;
+       /* Don't set the status_height for oldwin yet, this might break
+        * frame_fix_height(oldwin), therefore will be set below. */
    }
 #ifdef FEAT_VERTSPLIT
    if (flags & WSP_BOT)
@@ -1244,6 +1245,11 @@ win_split_ins(size, flags, new_wp, dir)
 #endif
    frame_fix_height(wp);
    frame_fix_height(oldwin);
+
+   if (!before)
+       /* new window above current one, set the status_height after
+        * frame_fix_height(oldwin) */
+       oldwin->w_status_height = STATUS_HEIGHT;
     }

     if (flags & (WSP_TOP | WSP_BOT))
```

### CODE REVIEW

**Comment by Pyrohh** [comment]
Are you able to reproduce this?

File: src/nvim/window.c [hunk 1]
```diff
@@ -911,12 +911,18 @@ int win_split_ins(int size, int flags, win_T *new_wp, int dir)
     } else {          /* new window below current one */
       wp->w_winrow = oldwin->w_winrow + oldwin->w_height + STATUS_HEIGHT;
       wp->w_status_height = oldwin->w_status_height;
-      oldwin->w_status_height = STATUS_HEIGHT;
+      /* Don't set the status_height for oldwin yet, this might break
+       * frame_fix_height(oldwin), therefore will be set below. */
```
Comment (glts) [inline src/nvim/window.c]: Use `//` style comments, may reflow lines to 80 col width.

File: src/nvim/window.c [hunk 2]
```diff
@@ -911,12 +911,18 @@ int win_split_ins(int size, int flags, win_T *new_wp, int dir)
...
     }
     if (flags & WSP_BOT)
       frame_add_statusline(curfrp);
     frame_fix_height(wp);
     frame_fix_height(oldwin);
+
+    if (!before)
```
Comment (glts) [inline src/nvim/window.c]: Curly brackets.

File: src/nvim/window.c [hunk 3]
```diff
@@ -911,12 +911,18 @@ int win_split_ins(int size, int flags, win_T *new_wp, int dir)
...
+      /* new window above current one, set the status_height after
```
Comment (glts) [inline src/nvim/window.c]: Shouldn't this be 'new window *below* the current one'?

**Comment by glts** [comment]
Repro instructions here https://groups.google.com/d/msg/vim_dev/fNq8tHTcKfA/7yg4GZ1h8agJ

LGTM, bug seems fixed. I do see a stray `~` in the command-line window, however. Going to look at it later.

**Comment by glts** [comment]
Yes, one line of the open vertically split buffers bleeds into the command-line window, but the bug is also in current Vim.

**Comment by buztard** [comment]
Updated according to your comments.
BTW, this is how it looks without the patch: 
vsplit

Fixed version: 
vsplit-2

File: src/nvim/window.c (hunk 3 above)
Comment (glts) [inline src/nvim/window.c]: bump

**Comment by Pyrohh** [comment]
Thanks @glts and @buztard, I can reproduce that. Merged in df3abf4df08606dd35126377ceec051763618836.

**Comment by glts** [comment]
mentioning @chrisbra /vim_dev

The bug visible in the second screenshot above is still present in Vim 7.4.712.

**Comment by chrisbra** [comment]
On Do, 30 Apr 2015, David Bürgin wrote:

> The bug visible in the second screenshot above is still present in Vim 7.4.712.

That is just a redraw glitch, isn't it?

Best,
Christian
-- 
Es gibt Dinge, die jeder leicht haben kann, nur bücken muß er sich
danach.
		-- Adalbert Stifter

**Pull request closed without merging**